cache/redis_client.go

```diff
@@ -0,0 +1,463 @@
+	// with one of the connections defined in the ACL list when connecting
+	// to a Redis 6.0 instance, or greater, that is using the Redis ACL system.
+	Username string `yaml:"username"`
+
+	// Optional password. Must match the password specified in the
+	// requirepass server configuration option (if connecting to a Redis 5.0 instance, or lower),
+	// or the User Password when connecting to a Redis 6.0 instance, or greater,
+	// that is using the Redis ACL system.
+	Password string `yaml:"password"`
+
+	// DB Database to be selected after connecting to the server.
+	DB int `yaml:"db"`
+
+	// DialTimeout specifies the client dial timeout.
+	DialTimeout time.Duration `yaml:"dial_timeout"`
+
+	// ReadTimeout specifies the client read timeout.
+	ReadTimeout time.Duration `yaml:"read_timeout"`
+
+	// WriteTimeout specifies the client write timeout.
+	WriteTimeout time.Duration `yaml:"write_timeout"`
+
+	// Maximum number of socket connections.
+	PoolSize int `yaml:"pool_size"`
+
+	// MinIdleConns specifies the minimum number of idle connections which is useful when establishing
+	// new connection is slow.
+	MinIdleConns int `yaml:"min_idle_conns"`
+
+	// Amount of time after which client closes idle connections.
+	// Should be less than server's timeout.
+	// -1 disables idle timeout check.
+	IdleTimeout time.Duration `yaml:"idle_timeout"`
+
+	// Connection age at which client retires (closes) the connection.
+	// Default 0 is to not close aged connections.
+	MaxConnAge time.Duration `yaml:"max_conn_age"`
+
+	// MaxAsyncConcurrency specifies the maximum number of SetAsync goroutines.
+	MaxAsyncConcurrency int `yaml:"max_async_concurrency"`
+
+	// MaxAsyncBufferSize specifies the queue buffer size for SetAsync operations.
+	MaxAsyncBufferSize int `yaml:"max_async_buffer_size"`
+
+	// MaxGetMultiConcurrency specifies the maximum number of concurrent GetMulti() operations.
+	// If set to 0, concurrency is unlimited.
+	MaxGetMultiConcurrency int `yaml:"max_get_multi_concurrency"`
+
+	// MaxItemSize specifies the maximum size of an item stored in redis.
+	// Items bigger than MaxItemSize are skipped.
+	// If set to 0, no maximum size is enforced.
+	MaxItemSize flagext.Bytes `yaml:"max_item_size"`
+
+	// GetMultiBatchSize specifies the maximum size per batch for mget.
+	GetMultiBatchSize int `yaml:"get_multi_batch_size"`
+
+	// TLSEnabled enable tls for redis connection.
+	TLSEnabled bool `yaml:"tls_enabled"`
+
+	// TLSConfig to use to connect to the redis server.
+	TLSConfig TLSConfig `yaml:"tls_config"`
+}
+
+func (c *RedisClientConfig) validate() error {
+	if c.Endpoint == "" {
+		return errMemcachedConfigNoEndpoint
+	}
+	// Set async only available when MaxAsyncConcurrency > 0.
+	if c.MaxAsyncConcurrency <= 0 {
+		return errRedisMaxAsyncConcurrencyNotPositive
+	}
+	return nil
+}
+
+func (c *RedisClientConfig) getAddresses() ([]string, error) {
+	addresses := strings.Split(c.Endpoint, ",")
```

Comment:
The CLI flag description doesn't explain these are comma separated. Should be mentioned.

Reply:
Also you should directly use `flagext.StringSliceCSV` in the config field, instead of a `string` which then is split.